media/js/base/stub-attribution.js

```diff
@@ -0,0 +1,264 @@
+            return url;
+        }
+
+        // append stub attribution query params.
+        $.each(data, function(key, val) {
+            if (key === 'attribution_code' || key === 'attribution_sig') {
+                url += (url.indexOf('?') > -1 ? '&' : '?') + key + '=' + val;
+            }
+        });
+
+        return url;
+    };
+
+    /**
+     * Handles XHR request from `stub_attribution_code` service.
+     * @param {Object} data - attribution_code, attribution_sig.
+     */
+    StubAttribution.onRequestSuccess = function(data) {
+        if (data.attribution_code && data.attribution_sig) {
+            // Update download links on the current page.
+            StubAttribution.updateBouncerLinks(data);
+            // Store attribution data in a cookie should the user navigate.
+            StubAttribution.setCookie(data);
+        }
+    };
+
+    /**
+     * AJAX request to bedrock service to authenticate stub attribution request.
+     * @param {Object} data - utm params and referrer.
+     */
+    StubAttribution.requestAuthentication = function(data) {
+        var SERVICE_URL = window.location.protocol + '//' + window.location.host + '/en-US/firefox/stub_attribution_code/';
+        $.get(SERVICE_URL, data).done(StubAttribution.onRequestSuccess);
+    };
+
+    /**
+     * Gets utm parameters and referrer information from the web page if they exist.
+     * @param {String} ref - Optional referrer to facilitate testing.
+     * @return {Object} - Stub attribution data object.
+     */
+    StubAttribution.getAttributionData = function(ref) {
+        var params = new window._SearchParams().utmParams();
+        var referrer = typeof ref !== 'undefined' ? ref : document.referrer;
+        var utmCount = 0;
+
+        for (var utm in params) {
+            if (params.hasOwnProperty(utm)) {
+                utmCount += 1;
+            }
+        }
+
+        // if there are no utm params and no referrer, do nothing.
+        if (utmCount === 0 && (typeof referrer === 'undefined' || referrer === '')) {
+            return false;
+        }
+
+        /* eslint-disable camelcase */
+        return {
+            utm_source: params.utm_source,
+            utm_medium: params.utm_medium,
+            utm_campaign: params.utm_campaign,
+            utm_content: params.utm_content,
+            referrer: referrer
+        };
+        /* eslint-enable camelcase */
+    };
+
+    /**
+     * Determine if the current page is scene2 of /firefox/new/.
+     * This is needed as scene2 auto-initiates the download. There is little point
+     * trying to make an XHR request here before the download begins, and we don't
+     * want to make the request a dependency on the download starting.
+     * @return {Boolean}.
+     */
+    StubAttribution.isFirefoxNewScene2 = function(location) {
+        location = typeof location !== 'undefined' ? location : window.location.href;
+        return location.indexOf('/firefox/new/?scene=2') > -1;
```

Comment:
Since we're dealing with external links, are we sure `scene=2` will always directly follow the `?`? Could we hit a scenario where `_utm` params are injected before the `scene=2` key/val pair?